src/node/coinstats.cpp

```diff
@@ -193,4 +187,37 @@ static void FinalizeHash(MuHash3072& muhash, CCoinsStats& stats)
     stats.hashSerialized = out;
 }
 static void FinalizeHash(std::nullptr_t, CCoinsStats& stats) {}
+
+std::optional<CCoinsStats> LookupUTXOStatsWithIndex(CoinStatsIndex& coin_stats_index, const CBlockIndex* pindex)
+{
+    CCoinsStats stats{Assert(pindex)->nHeight, pindex->GetBlockHash()};
+
+    stats.index_used = true;
+    if (!coin_stats_index.LookUpStats(pindex, stats)) {
+        return std::nullopt;
+    }
+
+    return stats;
+}
+
+std::optional<CCoinsStats> LookupUTXOStatsWithIndex(CoinStatsIndex& coin_stats_index, CCoinsView* view, BlockManager& blockman)
+{
+    CBlockIndex* pindex = WITH_LOCK(::cs_main, return blockman.LookupBlockIndex(view->GetBestBlock()));
+
+    return LookupUTXOStatsWithIndex(coin_stats_index, pindex);
+}
+
+std::optional<CCoinsStats> GetUTXOStats(CCoinsView* view, BlockManager& blockman, CoinStatsHashType hash_type, const std::function<void()>& interruption_point, const CBlockIndex* pindex, bool index_requested)
+{
+    // Use CoinStatsIndex if it is requested and available and a hash_type of Muhash or None was requested
+    if ((hash_type == CoinStatsHashType::MUHASH || hash_type == CoinStatsHashType::NONE) && g_coin_stats_index && index_requested) {
+        if (pindex) {
+            return LookupUTXOStatsWithIndex(*g_coin_stats_index, pindex);
+        } else {
+            return LookupUTXOStatsWithIndex(*g_coin_stats_index, view, blockman);
+        }
+    }
+
+    return ComputeUTXOStats(hash_type, view, blockman, interruption_point);
```

Comment:
In commit "coinstats: Separate hasher/index lookup codepaths" (e4b2969b54a3ddba507de1aeb17b6765ca1f102b)

Behavior precedes this PR, but it seems like `pindex` just gets ignored in this case. It might be nice to `assert(pindex == nullptr)` here or else have a comment "// pindex is ignored if coinstats index is not available or wasn't requested. Coinstats for the view are returned instead. This is ok because \[reasons]"

Reply:
Fixed in d03f4443e8deb477d0fd4b27056eaeec8a615ca3, thanks!